std/uni.d

```diff
@@ -7286,10 +7287,36 @@ if (isInputRange!Range && is(Unqual!(ElementType!Range) == Grapheme))
 }
 
 /// Ditto
-Range byCodePoint(Range)(Range range)
+auto byCodePoint(Range)(Range range)
 if (isInputRange!Range && is(Unqual!(ElementType!Range) == dchar))
 {
-    return range;
+    static if (isNarrowString!Range)
+    {
+        static struct Result
+        {
+            private Range _range;
+            @property bool empty() { return _range.empty; }
+            @property dchar front(){ return _range.front; }
+            void popFront(){ _range.popFront; }
+            static if (hasLength!Range)
+            {
+                @property size_t length(){ return _range.length; }
+            }
+            static if (isForwardRange!Range)
+            {
+                @property auto save() { return Result(_range.save); }
```

Comment:
This line is untested. Also, I don't think you need all the static if's since `isNarrowString` passes only for `char[]` and `wchar[]` derivatives -- we know what they are capable of. They don't hurt of course.

Reply:
Done